src/targets/objc/native.js

```diff
@@ -1,65 +1,84 @@
 'use strict';
 
 var util = require('util');
+var objcHelpers = require('./helpers');
 
 module.exports = function (options) {
   var opts = util._extend({
-    timeout: '10'
+    timeout: '10',
+    indent: true
   }, options);
 
   var code = [];
 
-  // Dependencies
   code.push('#import <Foundation/Foundation.h>');
+
+  if (this.source.cookies.length) {
+    var cookies = this.source.cookies.map(function (cookie) {
+      return encodeURIComponent(cookie.name) + '=' + encodeURIComponent(cookie.value);
+    });
+    this.source.headersObj['Cookie'] = cookies.join('; ');
+  }
+
+  // Build headers (including cookies)
+  if (Object.keys(this.source.headersObj).length) {
+    code.push(null);
+    code.push(objcHelpers.nsDictionaryBuilder('headers', this.source.headersObj, opts.indent));
+  }
+
+  var parameters;
+
+  if (this.source.postData.mimeType && (this.source.postData.text || this.source.postData.params)) {
+    switch (this.source.postData.mimeType) {
+      case 'application/x-www-form-urlencoded':
+        parameters = this.source.postData.params;
+
+        code.push(null);
+        code.push('NSMutableData *postData = [[NSMutableData alloc] initWithData:[@"' + parameters[0].name + '=' + parameters[0].value + '" dataUsingEncoding:NSUTF8StringEncoding]];');
+
+        for (var i = 1, len = parameters.length; i < len; i++) {
+          code.push('[postData appendData:[@"&' + parameters[i].name + '=' + parameters[i].value + '" dataUsingEncoding:NSUTF8StringEncoding]];');
+        }
+        break;
+
+      case 'application/json':
+        try {
+          parameters = JSON.parse(this.source.postData.text);
+          code.push(objcHelpers.nsDictionaryBuilder('parameters', parameters, opts.indent));
+          code.push(null);
+          code.push('NSData *postData = [NSJSONSerialization dataWithJSONObject:parameters options:0 error:nil];')
+        } catch (e) {}
+        break;
+
+      case 'multipart/form-data':
+        break;
+
+      default:
+        code.push(null);
+        code.push('NSData *postData = [[NSData alloc] initWithData:[@"' + this.source.postData.text.replace(/"/g, '\\"') + '" dataUsingEncoding:NSUTF8StringEncoding]];');
```

Comment:
I might be wrong ... will have to test different scenarios ... like what if the text was a JSON string already ...